### PR TITLE
Adding the ability to add constants to dictionaries and freshened up examples

### PR DESCRIPTION
**Pull request by samumartinf**

### CODE REVIEW

File: ode_composer/sbl.py [hunk 1]
```diff
@@ -98,19 +99,25 @@ def regularizer(self, w):
...
     def objective_fn(self, w):
-        return self.data_fit(w) + self.lambda_param * self.regularizer(w)
+        return self.data_fit(w) + 2*self.lambda_param * self.regularizer(w)
 
     def estimate_model_parameters(self):
         try:
             self.problem.solve(**self.solver_keywords)
 
             if self.config["solver"]["show_time"]:
+<<<<<<< HEAD
```
Comment (zoltuz) [inline ode_composer/sbl.py]: here some text remained when you resolved a merge conflict

File: examples/run_lotka_volterra.py [hunk 2]
```diff
@@ -12,17 +12,18 @@
 
 # define Lotka-Volerra model
 states = {"x1": "alpha*x1-beta*x1*x2", "x2": "delta*x1*x2-gamma*x2"}
-parameters = {"alpha": 2 / 3, "beta": 4 / 3, "delta": 1, "gamma": 1}
+parameters = {"alpha": 2 / 3, "beta": 4 / 3, "delta": 2, "gamma": 1}
```
Comment (zoltuz) [inline examples/run_lotka_volterra.py]: is there a reason why you have changed this example?

File: examples/run_sbl.py [hunk 3]
```diff
@@ -22,8 +22,10 @@
 y = np.array([2, 2, 2])
 
 lambda_param = 0.5
+linmodel = LinearModel(A, y, lambda_param)
 
-sbl = SBL(dict_mtx=A, data_vec=y, lambda_param=lambda_param)
+
+sbl = SBL(linear_model=linmodel)
```
Comment (zoltuz) [inline examples/run_sbl.py]: not sure this will work

File: ode_composer/.ipynb_checkpoints/measurements_generator-checkpoint.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+from ode_composer.statespace_model import StateSpaceModel
```
Comment (zoltuz) [inline ode_composer/.ipynb_checkpoints/measurements_generator-checkpoint.py]: can you add the .ipynp_checkpoints folder to .gitignore and removed this file?

File: ode_composer/.ipynb_checkpoints/signal_preprocessor-checkpoint.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
+from sklearn.gaussian_process import GaussianProcessRegressor
```
Comment (zoltuz) [inline ode_composer/.ipynb_checkpoints/signal_preprocessor-checkpoint.py]: same here, please remove it

File: ode_composer/dictionary_builder.py [hunk 6]
```diff
@@ -29,6 +29,15 @@ def evaluate_dict(self, input_data: Dict) -> np.ndarray:
             )
             # each dictionary function's weight gets a parameter name
             d_fcn.constant_name = f"w{idx+1}"
+
```
Comment (zoltuz) [inline ode_composer/dictionary_builder.py:32]: looks good to me!

File: ode_composer/measurements_generator.py [hunk 7]
```diff
@@ -38,9 +38,7 @@ def get_measurements(self, SNR_db=None):
             for idx, y in enumerate(self.sol.y):
                 Esym = np.sum(abs(y) ** 2) / len(y)
                 N_PSD = (Esym) / SNR
-                y_measured[idx, :] = y + np.sqrt(N_PSD) * np.random.randn(
-                    len(y)
-                )
+
```
Comment (zoltuz) [inline ode_composer/measurements_generator.py:44]: where did this piece of code go?

File: ode_composer/util.py [hunk 8]
```diff
@@ -49,6 +49,9 @@ def create_function(
             fcn_pointer=func,
             symbolic_expression=sym_expr,
             constant=weight,
+<<<<<<< HEAD
```
Comment (zoltuz) [inline ode_composer/util.py]: please remove this as well. It's a leftover from the conflict merging

File: tests/test_linear_model.py [hunk 9]
```diff
@@ -31,16 +31,3 @@ def test_linear_parameter_and_data_num():
 
     assert lin_model.parameter_num == lin_model.dict_mtx.shape[1]
     assert lin_model.data_num == lin_model.dict_mtx.shape[0]
-
```
Comment (zoltuz) [inline tests/test_linear_model.py:34]: is there a reason this test was removed?

**Review by zoltuz — CHANGES_REQUESTED** [review]
There are some blocking issues, please have a look at the comments below!

File: ode_composer/sbl.py (hunk 1 above)
Comment (samumartinf) [reply ode_composer/sbl.py]: This should be solved now

Comment (zoltuz) [reply ode_composer/sbl.py]: ok, thanks!

**Review by zoltuz — APPROVED** [review]
Looks good to me!

**Pull request merged**